From: Onigiri
Subject: [PATCH] Add a default-off toggle that shows the inner screen when open

The switch stores persist.sys.onigiri.inner_screen. With it on, services.jar shows the inner
panel while the phone is half or fully open, as on a stock Flip, from the next time it opens.

diff --git a/res/values-vi/strings.xml b/res/values-vi/strings.xml
--- a/res/values-vi/strings.xml
+++ b/res/values-vi/strings.xml
@@ -8073,4 +8073,5 @@
     <string name="onigiri_monitor_now_title">Màn hình</string>
     <string name="onigiri_monitor_level_title">Độ sáng</string>
     <string name="onigiri_monitor_auto_learns">Theo ánh sáng và học theo thay đổi của bạn</string>
+    <string name="onigiri_inner_screen_title">Dùng màn hình trong khi mở máy</string>
 </resources>
diff --git a/res/values/strings.xml b/res/values/strings.xml
--- a/res/values/strings.xml
+++ b/res/values/strings.xml
@@ -15442,4 +15442,5 @@
     <string name="onigiri_monitor_now_title">Monitor</string>
     <string name="onigiri_monitor_level_title">Brightness</string>
     <string name="onigiri_monitor_auto_learns">Follows the light and learns your changes</string>
+    <string name="onigiri_inner_screen_title">Use inner screen when open</string>
 </resources>
diff --git a/res/xml/onigiri_settings.xml b/res/xml/onigiri_settings.xml
--- a/res/xml/onigiri_settings.xml
+++ b/res/xml/onigiri_settings.xml
@@ -28,6 +28,7 @@
         <SecSwitchPreference android:title="@string/onigiri_cover_keep_rotation_title" android:key="onigiri_cover_keep_rotation" settings:controller="moe.foss.onigiri.settings.CoverKeepRotationPreferenceController" />
     </SecPreferenceCategory>
     <SecPreferenceCategory android:title="@string/onigiri_display_settings_title">
+        <SecSwitchPreference android:title="@string/onigiri_inner_screen_title" android:key="onigiri_inner_screen" settings:controller="moe.foss.onigiri.settings.InnerScreenPreferenceController" />
         <SecSwitchPreference android:title="@string/onigiri_max_refresh_rate_title" android:key="onigiri_max_refresh_rate" settings:controller="moe.foss.onigiri.settings.ForceMaxRefreshRatePreferenceController" />
         <SecSwitchPreference android:title="@string/onigiri_dex_high_refresh_title" android:key="onigiri_dex_high_refresh" settings:controller="moe.foss.onigiri.settings.DexHighRefreshPreferenceController" />
         <SecSwitchPreference android:title="@string/onigiri_mirror_high_refresh_title" android:key="onigiri_mirror_high_refresh" settings:controller="moe.foss.onigiri.settings.MirrorHighRefreshPreferenceController" />
diff --git a/smali_classes4/moe/foss/onigiri/settings/InnerScreenPreferenceController.smali b/smali_classes4/moe/foss/onigiri/settings/InnerScreenPreferenceController.smali
new file mode 100644
--- /dev/null
+++ b/smali_classes4/moe/foss/onigiri/settings/InnerScreenPreferenceController.smali
@@ -0,0 +1,48 @@
+.class public Lmoe/foss/onigiri/settings/InnerScreenPreferenceController;
+.super Lcom/android/settings/core/TogglePreferenceController;
+.source "InnerScreenPreferenceController.java"
+
+.method public constructor <init>(Landroid/content/Context;Ljava/lang/String;)V
+    .locals 0
+    invoke-direct {p0, p1, p2}, Lcom/android/settings/core/TogglePreferenceController;-><init>(Landroid/content/Context;Ljava/lang/String;)V
+    return-void
+.end method
+
+.method public getAvailabilityStatus()I
+    .locals 0
+    const/4 p0, 0x0
+    return p0
+.end method
+
+.method public getSliceHighlightMenuRes()I
+    .locals 0
+    sget p0, Lcom/android/settings/R$string;->menu_key_onigiri_settings:I
+    return p0
+.end method
+
+.method public isChecked()Z
+    .locals 1
+    const-string p0, "persist.sys.onigiri.inner_screen"
+    const/4 v0, 0x0
+    invoke-static {p0, v0}, Landroid/os/SemSystemProperties;->getBoolean(Ljava/lang/String;Z)Z
+    move-result p0
+    return p0
+.end method
+
+.method public setChecked(Z)Z
+    .locals 1
+    :try_start
+    const-string p0, "persist.sys.onigiri.inner_screen"
+    invoke-static {p1}, Ljava/lang/Boolean;->toString(Z)Ljava/lang/String;
+    move-result-object v0
+    invoke-static {p0, v0}, Landroid/os/SemSystemProperties;->set(Ljava/lang/String;Ljava/lang/String;)V
+    :try_end
+    .catch Ljava/lang/IllegalArgumentException; {:try_start .. :try_end} :failed
+    .catch Ljava/lang/SecurityException; {:try_start .. :try_end} :failed
+    const/4 p0, 0x1
+    return p0
+    :failed
+    move-exception p0
+    const/4 p0, 0x0
+    return p0
+.end method
